pkg/domain/entities/pods.go

```diff
@@ -266,6 +266,15 @@ type ContainerCreateOptions struct {
 	CgroupConf []string
 }
 
+func NewInfraContainerCreateOptions() ContainerCreateOptions {
```

Comment:
I like the idea of this, but would it make more sense to make this a struct rather than a function? A struct would more closely resemble our other create options while a function more closely represents our specgenerator process. either way this is a good way to differentiate infra.

Reply:
A new struct for an infra container would be helpful to clarify what parameter is required for creating an infra container. But, I'm not sure how much change is necessary to introduce the new struct. I chose a function to fix this issue with the smaller change.